### DISCUSSION TITLE
Unable to update bound field from a json column

### DISCUSSION THREAD

**Discussion by abishekrsrikaanth** [livewire/livewire#3847] [1. Help]
When updating the value on a json column through Model binding, I get this error 

```
Indirect modification of overloaded element of ModelName has no effect.
```

FYI @joshhanley, as discussed on discord, referring you here.

Here is a sample code for your reference.

**Model**

```php
public class Foo extends Model
{
   protected $casts = ['bar'=>'array'];
}
```

**Livewire Component**

```php

use App\Models\Foo;

public class FooComponent extends Component
{
    public Foo $foo;

    protected $rules = [
        'foo.bar.name'=>['required']
    ];

    public function mount()
    {
          $this->foo = Foo::find(1);
    }
}
```

**Blade View**

```html

<input type="number" wire:model.lazy="foo.bar.name" />

```

**Comment by terry-greenlaw** [comment]
I wasn't able to find a way to directly manipulate the bound json column
through the array cast, but a copy/modify worked.

   - assign the json column to an array
   - make your changes to the array
   - assign the array back to the json column

HTH

**Comment by ighmouraceneb** [comment]
@abishekrsrikaanth  We will not be able to modify directly the json attributes in laravel : 
[Éloquent : Mutateurs & Casting](https://laravel.com/docs/8.x/eloquent-mutators#array-and-json-casting)
